diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -0,0 +1,118 @@
+    private var fileName: String? = null
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        fileName = params.fileName
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        when (node.elementType) {
+            FUN -> checkFun(node)
+            IF -> checkIf(node)
+            WHEN -> checkWhen(node)
+            FOR -> checkFor(node)
+        }
+    }
+
+    private fun checkFor(node: ASTNode) {
+        checkOpenBrace(node, BODY)
+        checkCloseBrace(node.findChildByType(BODY)!!.findChildByType(BLOCK)!!)
+    }
+
+    private fun checkWhen(node: ASTNode) {
+        checkOpenBrace(node, LBRACE)
+        checkCloseBrace(node)
+    }
+
+    private fun checkFun(node: ASTNode) {
+        if (node.hasChildOfType(BLOCK)) {
+            checkOpenBrace(node, BLOCK)
+            checkCloseBrace(node.findChildByType(BLOCK)!!)
+        }
+    }
+
+    private fun checkIf(node: ASTNode) {
+        val ifPsi = node.psi as KtIfExpression
+        val thenNode = ifPsi.then?.node
+        val hasElseBranch = ifPsi.elseKeyword != null
+        val elseNode = ifPsi.`else`?.node
+
+        checkOpenBrace(node, THEN)
+        checkCloseBrace(thenNode!!)
+        if (hasElseBranch && elseNode!!.elementType != IF) {
+            checkOpenBrace(node, ELSE)
+            checkCloseBrace(elseNode)
+        }
+    }
+
+    private fun checkOpenBrace(node: ASTNode, beforeType: IElementType) {
+        val braceSpace = node.findChildBefore(beforeType, WHITE_SPACE)
+        if (!(braceSpace == null || braceSpace.elementType != WHITE_SPACE)) {
+            if (braceSpace.text.contains("\n".toRegex())) {
+                BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "open braces", 0) {
+                }
+            }
+        }
+        val nodeText = when (node.elementType) {
+            IF -> {
+                when (beforeType) {
+                    THEN -> node.findChildByType(THEN)?.text
+                    else -> node.findChildByType(ELSE)?.text
+                }
+            }
+            WHEN -> node.findChildBefore(WHEN_ENTRY, WHITE_SPACE)?.text + node.getFirstChildWithType(WHEN_ENTRY)?.text
+            FOR -> node.findChildByType(BODY)?.text
+            else -> node.findChildByType(BLOCK)?.text
+        }
+        if (nodeText != null && nodeText[1] != '\n') {
+            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "open braces", 0) {
+            }
+        }
+    }
+
+    private fun checkCloseBrace(node: ASTNode) {
+        val space = node.getChildren(null).findLast { it.elementType == WHITE_SPACE }
+        if (!space!!.text.contains("\n".toRegex())) {
```

Comment:
```suggestion
        val space = node.getChildren(null).findLast { it.elementType == WHITE_SPACE }!!
        if (!space.text.contains("\n".toRegex())) {
```
I think this is more readable: if you are sure this variable is not null, you can as well check it right in the asignment.